### ISSUE TITLE
Gate of the Fold needs documentation

### ISSUE THREAD

**Issue by Deiarch** [WayofTime/BloodMagic#1440]
#### Issue Description:
I can't teleport with Gate of the Fold.

#### What happens:
I'm building the rituals using ritual stones, and Diviner (dusk). I've initially used some stone bricks as the "key," but I also attempted this with no key. I'm activating them with a weak activation crystal. Upon entering the portal, I simply pass straight through it instead of teleporting to its identical pair.

I tested it further by getting the latest forge version, with only the Blood Magic and Guide API mods installed. Same results.

#### What you expected to happen:
I expected to teleport from one portal to the other. I didn't. 

I'm not 100% sure, but it appears I may be teleporting (at least some of the time) to the center of whichever portal I enter. Perhaps the portal is linking to itself?

#### Steps to reproduce:

1. Build a pair of Gates of the Fold
2. Activate them
3. Attempt to teleport
...

____
#### Affected Versions (Do *not* use "latest"):

- BloodMagic: 1.12.2-2.3.2-100
- Minecraft: 1.12.2
- Forge: tested in 2745 and 2760

**Comment by keraldi** [comment]
Gate of the Fold is not exactly builder-friendly.
If you built it on grass, chances are some grass turned to dirt so the key changed.
You might also have mirrored the key by accident or build the Gate of the Fold in the wrong direction towards the key.

I'll test it myself again to see if it works.

**Comment by Deiarch** [comment]
What are the dimensions of the key? What I mean is, how far out does the code look, relative to the MRS?

**Comment by keraldi** [comment]
Pretty sure it's 1 block in every direction, including down, from the row/line the MRS is placed in.

**Comment by Deiarch** [comment]
Ok, I figured out my problems. I had initially built on top of grass, which caused the problem with some converting to dirt. My second problem was that I either had too many attempted gates with the same key (stone brick surrounding the gate at the same y-level as the MRS) or it was confused because I had stone brick on both sides of the gate. Either way, when I made two gates that had stone brick everywhere except one spot, they now work properly.

So, please consider this bug report changed to a documentation request. Thanks for the help!

**Comment by keraldi** [comment]
Then please be a dear and change the title and/or description and I'm pretty sure it says that somewhere <_<

**Comment by Deiarch** [comment]
After several more hours of testing, I'm pretty sure that the portal looks 2 blocks in every direction.

**Comment by MentalMouse** [comment]
So, 125 blocks including the MRS and some air/portal blocks?

**Comment by Deiarch** [comment]
The portal is 5x5x1, so if you add 2 blocks in every direction it's 9x9x5, which is 405 blocks. Depending on where you put it, many of those could be air.

**Comment by MentalMouse** [comment]
So all blocks within 2 blocks of the whole portal, not just the MRS?!    Well, that effectively means that if one end is free-standing, the other needs to be too, likewise for embedded in rock... does water count?  Hmm, are these meant to be able to go between dimensions?  If so, making a link to the Nether gets a little more interesting....

**Comment by VadersFist0501** [comment]
Much of Blood Magic is in need of documentation, I am afraid.

**Comment by keraldi** [comment]
Feel free to read the source code and/or create documentation. I'll gladly explain every mechanic into detail but I am too impatient to get into fancy wiki formatting.

**Comment by MentalMouse** [comment]
Never mind the Wiki (though the FTB pages are fairly out of date and based on their modpack), I'd like to see documentation of such things in the mod's own Guide.  (The Alchemy Table is completely undocumented, JEI notwithstanding!)

**Comment by keraldi** [comment]
Guide book documentation takes a lot of effort, especially since Way intends it to be interwoven with lore.
You shouldn't take mod documentation (from the mod itself or by the community) for granted, especially not because it's FOSS (which means the source code itself can be used as a reference, a big effort is put into its readability and tidiness).
Not to mention that missing documentation might be by design for some functionality (hidden features/easter eggs) or even a mod author's philosophy (self-explainatory mods/mods made to be highly intuitive and or miniscule tweaks that are to be discovered on the fly).
Of course that's not the case here but the mod is still in beta and it's a project being worked on in free time.

JEI is part of the mod documentation (highly recommended) and is to be used to look up recipes for the mod.
FTB wiki pages (both official and unofficial) can be edited by anyone and their main purpose is community-driven mod documentation. There's no single wiki page that describes functionality for just a single modpack as base functionality is always the main focus because 90% of the mod packs out there don't touch mod interaction.

Excuse myself, as I got a bit carried away while writing this.
After all, you put in but a simple request.

**Comment by MentalMouse** [comment]
On the one hand I understand that documentation takes effort.  I also suspect that Way is working on restoring the old Tier 6 behind the scenes which is presumably a big project.  On the other, I used to be a programmer ("in another lifetime"), and my own philosophy was always that documentation is part of the project.  So it frustrates me to see someone shorting it when the mod is otherwise fairly stable and feature-complete to a reasonable breakpoint (that is, Tier 5).

**Comment by keraldi** [comment]
Of course documentation is essential one way or the other, in some cases both (code documentation is always essential unless you want to keep your job or inflate your ego [security by obscurity]).

I tried myself with bringing a couple wiki pages up to date but it was too boring, tiresome and in my eyes unasked for and unneeded, as I find the source code pretty readable for most things needed and JEI as well as similarities to previous versions made the rest understandable; not to mention that it was only plain text and some tables. People will keep asking for demon summoning even if you plaster it everywhere and that should be example enough that even with top of the notch documentation people will ask questions or for more documentation.

Points are:
- Effort/Scope. Some people can design and program and whatnot. I (arguably) program but definitely suck at designing and therefore won't try to make wiki pages anymore.
Someone that is very familiar with the mod but has little time or is lazy (such as myself) might have all the knowledge you would want but not the time to compile it in a single place or would discard half of it because it seems obvious. The best way to realize it would be a couple people asking quick questions for answers that are then tested and compiled into a wiki. That would be faster and have higher quality than people without time and/or motivation doing the time-heavy testing and wiki formatting.

- Design/Aesthetics. I'm good enough for plain text, tables and some simple text formatting so I can **boldly** tell people to do my bidding.
If I keep seeing replies here I'll just write down an abomination of plain text in the wikis about the stuff as an example. You'll probably start seeing people gouge their eyes out because of their magnificence. I don't have the time nor is there any benefit for me in aquiring the skills or patience in making stuff look nice (it should be obvious that this is a foul lie and there are many upsides in being able to make stuff look nice. People might argue that I am indeed able to make stuff look nice but they are all liars, heretics and evil beings that just want to watch me suffer).

- Actuality/Up-to-date-ness/Currentness/other_words_that_sound_stupid_for_me_but_apparently_exist. Stuff changes. Every time stuff changes the wikis would need to be changed as well. See Effort/Scope. The existing documentation has been out of date since it got introduced.

- Motivation. Only somoene very dedicated to documentation will willingly allocate time for documentation only. The important parts of documentation are done on the fly for stuff that might not seem apparent (in the code).

- Leeches. In a perfect world ~~I could sleep all day.~~ every answered question would be compiled by the person that asked it so other people would find it and don't have to ask. And or the other people would have the decency to search first.
See "Tier 6", "demon invasion", "bound armor" and "I found a solution to my question but won't tell" phenomena, also egoism, people being embarassed because they think the solution was too obvious and people doing the same mistake twice and stumbling on their own question asked and answered with "NVM, I found it" on StackOverflow/StackExchange.

Also keep these things in mind (if stuff has been repeated this is intentional for learning purposes and time savings as I don't have an editor):

Documentation of the source itself/readability and maintainability of the source code has highest priority (along with performance and intended behavior/bug-free-ness).
There is no release version for the most recent/playable/current mod version.
Documentation takes way more effort/is much more bothersome and feels less rewarding/accomplished than creating or fixing features of the mod.
Most people manage to figure the holes in the mod documentation out.
There are existing community driven mod documentations (which are most of the time preferrable because of filtered and expanded information, interactions, trivia etc., everyone can take part and in fact instead of discussing here such articles could have been written but people chose not to)...

The discord server is always available for questions and over and over again people prove that documentation won't necessarily eliminate the need for documentation.
There are people that don't look at the in-game documentation or at the wikis which provide a guideline.
Meanwhile most questions are answered paitently and nobody stops the people that ask those questions to compile them together in a wiki.

And again: it's easier for a couple dozen people to sacrifice an hour each for a wiki page or even just a couple paragraphs than for one or two people to sacrifice days and days for the same thing.

We all do this in our free time. If you want to enjoy something, you might want to take care about it and invest some time into it.
